### ISSUE TITLE
Only the terminal node of a PRT shows its feedback

### ISSUE THREAD

**Issue by alcarola** [maths/moodle-qtype_stack#940]
I created a multilanguage multipart question with a multistep PRT (attached). 

My problem is that only the terminal node of the PRT displays its feedback. I want all the nodes to display their feedback. See the attached testing protocol (and the question preview behaves the same, only displaying the feedback of the terminal node).

[Testing question Nonlinear algebraic system [FI_EN_SV].pdf](https://github.com/maths/moodle-qtype_stack/files/11171649/Testing.question.Nonlinear.algebraic.system.FI_EN_SV.pdf)

[questions-NumMat 2023-Nonlinear algebraic system [FIENSV]-20230406-1735.xml.zip](https://github.com/maths/moodle-qtype_stack/files/11171661/questions-NumMat.2023-Nonlinear.algebraic.system.FIENSV.-20230406-1735.xml.zip)

**Comment by aharjula** [comment]
I think we have seen this before. If you turn the multi-lang filter off, you should see that that feedback does indeed come out. It is just the way the multi-lang filter deals with spans that are immediately next to each other, leading to just the last one being outputted. By next to each other, I mean without a particular type of white space between the tags. Possibly newlines are different than spaces or some such thing. I think that people have been adding non-visible symbols to deal with this but I could not quickly figure out a simple solution.

We could probably make whatever extra is needed to be added between the feedback from the nodes, but that might affect the layout for some users.

It would be nice if the actual multi-lang filter were fixed, but while waiting for that, have you tried the `[[lang]]`-block? Basically, these are largely equivalent, since 4.4.2 + the fixes listed in #905:

```
<span class="multilang" lang="EN">...</span> = [[lang code="EN"]]...[[/lang]]
```
This gets processed entirely inside STACK and does not tie up any span-tags thus avoiding some annoying nesting issues.

**Comment by alcarola** [comment]
Thanks a lot for your input! I by the way use STACK version 2022082900.

I tried your suggestion and discovered a bunch of problems; see the attached PDF (which uses the language setting Swedish (Finland):

a) The selected language is displayed only in the question text, but not in the general feedback (model solution) or PRT feedback. In the latter, the English text is displayed regardless of Moodle language. 

b) The STACK implementation of multilanguage questions does not understand that Finland Swedish (sv_fi) should default to Swedish (sv) translations in case there are no sv_fi-specific translations. Hence, in the attached PDF, the question text is completely missing, because I used [[lang code="SV"]]. (Some of our students probably use the Sweden Swedish translations sv, even if they are of poor quality and largely unmaintained.) At the very least, the filter should default to some language.

c) Even if the new version of the question is still unusable for me, it was a horrifying amount of work to correct even one question, and we have a lot of multilang questions. The substitution that you gave above seems simple to implement in STACK, similar to the $ $ -> \( \) filter that is already there. Would it be possible to get an implementation of this?

I also attach the latest version of the file.

[Testar fråga Nonlinear algebraic system [FI_EN_SV].pdf](https://github.com/maths/moodle-qtype_stack/files/11210889/Testar.fraga.Nonlinear.algebraic.system.FI_EN_SV.pdf)

[frågor-NumMat 2023-Nonlinear algebraic system [FIENSV]-20230412-1433.xml.zip](https://github.com/maths/moodle-qtype_stack/files/11210924/fragor-NumMat.2023-Nonlinear.algebraic.system.FIENSV.-20230412-1433.xml.zip)
